build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/RunParallelCmds.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Diagnostics;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Tasks = System.Threading.Tasks;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public class RunParallelCmds : Task
+	{
+		[Required]
+		public ITaskItem[] Commands { get; set; }
+
+		public string ManagedRuntime { get; set; }
+
+		public string ManagedRuntimeArguments { get; set; }
+
+		public override bool Execute ()
+		{
+			bool success = true;
+
+			Tasks.Parallel.ForEach (Commands, new Tasks.ParallelOptions () { MaxDegreeOfParallelism = Environment.ProcessorCount },
+				cmd => {
+					var useManagedRuntime = !string.IsNullOrEmpty (ManagedRuntime);
+					var argumentsBeginning = useManagedRuntime ? ManagedRuntimeArguments + " " : "";
+					var procStartInfo = new ProcessStartInfo () {
+						FileName =  useManagedRuntime ? ManagedRuntime : cmd.GetMetadata ("Command"),
+						Arguments = $"{argumentsBeginning}{cmd.GetMetadata ("Arguments")}",
+						CreateNoWindow = true,
+						UseShellExecute = false,
+						RedirectStandardOutput = true,
+						RedirectStandardError = true,
+					};
+
+					Log.LogMessage (MessageImportance.Normal, $"Starting Command: {cmd.GetMetadata ("Command")} Arguments: {cmd.GetMetadata ("Arguments")}");
```

Comment:
don't use log. here. it will lock up the UI thread (if someone decides to build in the UI). 
You should add a reference to the Xamarin.AsyncTask Nuget, and derive from that. Then use the LogMessage overloaded methods to log any data. It will marshal stuff back to the UI thread correctly